src/picture.rs

```diff
@@ -238,39 +239,45 @@ pub fn rav1d_picture_copy_props(
 // itut_t35 was taken out of the c.itut_t35 originally, but that violates Rust
 // borrowing rules so we need to pass it to this function explicitly.
 pub(crate) fn rav1d_thread_picture_alloc(
-    c: &Rav1dContext,
+    fc: &Box<[Rav1dFrameContext]>,
+    logger: &Option<Rav1dLogger>,
+    allocator: &Rav1dPicAllocator,
+    content_light: &Option<Arc<Rav1dContentLightLevel>>,
+    mastering_display: &Option<Arc<Rav1dMasteringDisplay>>,
+    output_invisible_frames: bool,
+    frame_flags: &Atomic<PictureFlags>,
     f: &mut Rav1dFrameData,
     bpc: u8,
     itut_t35: Arc<Mutex<Vec<Rav1dITUTT35>>>,
 ) -> Rav1dResult {
     let p = &mut f.sr_cur;
-    let have_frame_mt = c.fc.len() > 1;
+    let have_frame_mt = fc.len() > 1;
     let frame_hdr = &***f.frame_hdr.as_ref().unwrap();
     picture_alloc_with_edges(
-        &c.logger,
+        logger,
         &mut p.p,
         frame_hdr.size.width[1],
         frame_hdr.size.height,
         f.seq_hdr.clone(),
         f.frame_hdr.clone(),
         bpc,
-        &c.allocator,
+        allocator,
     )?;
 
     rav1d_picture_copy_props(
         &mut p.p,
-        c.content_light.clone(),
-        c.mastering_display.clone(),
+        content_light.clone(),
+        mastering_display.clone(),
```

Comment:
Can we move the `.clone()`s to the caller and pass the `Arc`s by value?  `.clone()`ing in the caller is generally more flexible.

Reply:
Sure, done.